### PR TITLE
Switch to release-bot to handle releases

### LINKED ISSUE #123: Update to using release-bot
I'm holding off on doing the switch until the FreeBSD support PR is merged.

This one is a little different from the others that have been done because as part of the process in `release.yml` before we trigger the announcement, we will need to add a `latest-release` tag that is the same as the tag that triggered our release.

### PR DESCRIPTION
**Pull request by SeanTAllen**
Closes #123

### CODE REVIEW

**Comment by SeanTAllen** [comment]
macOS failure is expected at the moment.

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -49,15 +49,29 @@ jobs:
...
 
+  update-latest-release-tag:
+    name: Update latest-release tag
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v1
+      - name: Tag
+        run: |
+          git tag --force latest-release
+          git push --force "https://${TOKEN}@github.com/${GITHUB_REPOSITORY}.git" latest-release
```
Comment (Theodus) [inline .github/workflows/release.yml:60]: where does `GITHUB_REPOSITORY` come from?

Comment (SeanTAllen) [reply .github/workflows/release.yml:60]: Its defined by GH for every CI run.

See https://help.github.com/en/actions/configuring-and-managing-workflows/using-environment-variables

That series of Gitness right there was in the existing release script that is going away so we can use the more general bot.

File: .ci-scripts/release/start-a-release.bash [hunk 2]
```diff
@@ -1,127 +0,0 @@
...
-
-# push to release to remote
-echo -e "\e[34mPushing commited changes back to master\e[0m"
-git push ${PUSH_TO} master
-echo -e "\e[34mPushing ${VERSION} tag\e[0m"
-git push ${PUSH_TO} "${VERSION}"
-
-# add special "latest-release" tag that we move to point to latest release
-# this tag is what we have people use to install ponyup-init.sh via curl
-git tag --force latest-release
```
Comment (SeanTAllen) [inline .ci-scripts/release/start-a-release.bash:107]: see here @Theodus

File: .ci-scripts/release/start-a-release.bash [hunk 3]
```diff
@@ -1,127 +0,0 @@
...
-# no unset variables allowed from here on out
-# allow above so we can display nice error messages for expected unset variables
-set -o nounset
-
-# Set up .netrc file with GitHub credentials
-git config --global user.name 'Ponylang Main Bot'
-git config --global user.email 'ponylang.main@gmail.com'
-git config --global push.default simple
-
-PUSH_TO="https://${RELEASE_TOKEN}@github.com/${GITHUB_REPOSITORY}.git"
```
Comment (SeanTAllen) [inline .ci-scripts/release/start-a-release.bash:69]: And also here @Theodus

**Pull request merged**